src/pool/srv_pool.c

```diff
@@ -1235,8 +1235,7 @@ pool_svc_step_up_cb(struct ds_rsvc *rsvc)
 		if (svc->ps_pool != NULL)
 			fini_svc_pool(svc);
 	}
-	if (map_buf != NULL)
-		D_FREE(map_buf);
```

Comment:
Not a fan of these changes, or functions like this in general:

    int f(void *p)
    {
        if (p == NULL)
            return 0;
        ...
    }

The caller shouldn't have called f with NULL in the first place.

Reply:
In general I agree (in fact I made comment on daos_prop_free() saying exactly that on this PR), but free() accepts NULL, and we made a decision years ago to use D_FREE unconditionally as it's easier to scan the code and the extra debugging lines are useful.

That said, these changes were all made by a pre-commit hook I'm testing and aren't relevant to the leaks so I'll back them all out.

Reply:
I would also agree that check once inside the function is more preferable rather check everywhere in code before the call.